mlkem/asm/aarch64/rej_uniform_asm_clean.S

```diff
@@ -0,0 +1,585 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*************************************************
+* Name:        rej_uniform_asm
+*
+* Description: Full SIMD lane, run rejection sampling on uniform random bytes to generate
+*              uniform random integers mod q
+*
+* Arguments:   - int16_t *r:          pointer to output buffer
+*              - unsigned int len:    requested number of 16-bit integers
+*                                     (uniform mod q)
+*              - const uint8_t *buf:  pointer to input buffer
+*                                     (assumed to be uniform random bytes)
+*              - unsigned int buflen: length of input buffer in bytes
```

Comment:
There seem to be some assumptions about the buflen here, right? 
Could you document those please? 
Also if the assumption is that buflen is divisible by 24 and is at least 48, then maybe it's better to pass it in as a number of 24-byte blocks? That would make it harder to misused this code and make it a little bit easier to read.

Reply:
I rewrote things so that the buffer length can be anything (hopefully...)